classes/class-wc-connect-jetpack.php

```diff
@@ -159,40 +115,57 @@ public static function tracks_record_event( $user, $event_type, $data ) {
 		 * @return bool Whether or nor the current user is connected to Jetpack
 		 */
 		public static function is_current_user_connected() {
-			if ( class_exists( '\Automattic\Jetpack\Connection\Manager' ) && method_exists( '\Automattic\Jetpack\Connection\Manager', 'is_user_connected' ) ) {
-				$connection = new Manager();
-
-				return $connection->is_user_connected();
-			}
-
-			if ( defined( 'JETPACK_MASTER_USER' ) ) {
-				$user_token = self::get_master_user_access_token( JETPACK_MASTER_USER );
-
-				return ( isset( $user_token->external_user_id ) && get_current_user_id() === $user_token->external_user_id );
-			}
-
-			return false;
+			return self::get_connection_manager()->is_user_connected();
 		}
 
 		/**
-		 * Determines if Jetpack is connected
+		 * Determines if Jetpack is connected for the plugin
 		 *
 		 * @return bool Whether or nor Jetpack is connected
 		 */
 		public static function is_connected() {
-			if ( class_exists( '\Automattic\Jetpack\Connection\Manager' ) && method_exists( '\Automattic\Jetpack\Connection\Manager', 'is_connected' ) ) {
-				$connection = new Manager();
-
-				return $connection->is_connected();
-			}
+			$manager = self::get_connection_manager();
 
-			if ( defined( 'JETPACK_MASTER_USER' ) ) {
-				$user_token = self::get_master_user_access_token( JETPACK_MASTER_USER );
+			return $manager->is_plugin_enabled() && self::is_active() && ! $manager->is_missing_connection_owner();
+		}
 
-				return isset( $user_token->external_user_id );
+		/**
+		 * Connects the site to Jetpack.
+		 * This code performs a redirection, so anything executed after it will be ignored.
+		 *
+		 * @param $redirect_url
+		 */
+		public static function connect_site( $redirect_url ) {
+			// Mark the plugin as enabled in case it had been soft-disconnected.
+			$jetpack_connection_manager = self::get_connection_manager();
+			$jetpack_connection_manager->enable_plugin();
```

Comment:
`enable_plugin()` is also deprecated.

We used those `enable_plugin()`, `is_plugin_enabled()`, and `disable_plugin()` methods to "softly disconnect" plugins, meaning plugins could "disconnect" from WPCOM and leave the connection intact for all other connected plugins.
We no longer do that, as individual plugins cannot disconnect on their own (except for Jetpack), and "My Jetpack" is the main connection management place.
Here's some context: p9dueE-4n9-p2